kyo-core/shared/src/main/scala/kyo/Duration.scala

```diff
@@ -51,6 +51,39 @@ extension (value: Long)
     inline def weeks: Duration   = value.as(Weeks)
     inline def months: Duration  = value.as(Months)
     inline def years: Duration   = value.as(Years)
+    inline def nano: Duration    = compiletime.error("please use `.nanos`")
+    inline def micro: Duration   = compiletime.error("please use `.micros`")
+    inline def milli: Duration   = compiletime.error("please use `.millis`")
+    inline def second: Duration  = compiletime.error("please use `.seconds`")
+    inline def minute: Duration  = compiletime.error("please use `.minutes`")
+    inline def hour: Duration    = compiletime.error("please use `.hours`")
+    inline def day: Duration     = compiletime.error("please use `.days`")
+    inline def week: Duration    = compiletime.error("please use `.weeks`")
+    inline def month: Duration   = compiletime.error("please use `.months`")
+    inline def year: Duration    = compiletime.error("please use `.years`")
+end extension
+
+extension (value: 1)
+    inline def nano: Duration    = value.asNanos
+    inline def micro: Duration   = value.as(Micros)
+    inline def milli: Duration   = value.as(Millis)
+    inline def second: Duration  = value.as(Seconds)
+    inline def minute: Duration  = value.as(Minutes)
+    inline def hour: Duration    = value.as(Hours)
+    inline def day: Duration     = value.as(Days)
+    inline def week: Duration    = value.as(Weeks)
+    inline def month: Duration   = value.as(Months)
+    inline def year: Duration    = value.as(Years)
+    inline def nanos: Duration   = compiletime.error("please use `.nano`")
+    inline def micros: Duration  = compiletime.error("please use `.micro`")
+    inline def millis: Duration  = compiletime.error("please use `.milli`")
```

Comment:
I would prefer just not to populate these, I think the plural version should work for all values

Reply:
Ok! Agreed it's too much.